lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -406,7 +408,20 @@ void StagesSubject::stageChanged(
 
 #ifdef UFE_V2_FEATURES_AVAILABLE
         if (sendValueChangedFallback) {
-            valueChanged(ufePath, changedPath.GetNameToken());
+
+            // check to see if there is an entry which Ufe should notify about.
+            std::vector<const SdfChangeList::Entry*> entries = it.base()->second;
+            for (const auto& entry : entries)
+            {
+                // Adding an inert prim means we created a primSpec for an ancestor of
+                // a prim which has a real change to it.
+                if (entry->flags.didAddInertPrim || entry->flags.didRemoveInertPrim)
```

Comment:
This is the new filtering that catches the over primSpec creation & deletion. There are loads of other flags which could be relevant to check here.